### PR TITLE
Fix searching for tx2a only nodes in subnodes (#10)

### PR DESCRIPTION
**Pull request by m3harvey**
Cherry-picked from 0a8db2f10c2b

The earlier algorithm only checked the top level device tree nodes,
but this one recurses and checks the subnodes as well.
This is in preparation for removing the temperature nodes on tx2b

Done to help HW-3736

### CODE REVIEW

**Review by MikeLoder — DISMISSED** [review]

File: board/miovision/royaloak-smartsense/royaloak-smartsense.c [hunk 1]
```diff
@@ -200,57 +198,65 @@ static void ft_mac_address_setup(void *fdt)
...
-		--current_subnode_index;
-		name = subnodes_to_delete_names[current_subnode_index];
-		subnode = subnodes_to_delete[current_subnode_index];
-		debug("%s: deleting node %s\n", __func__, name);
-		del_ret = fdt_del_node(blob, subnode);
-		if (del_ret < 0) {
-			pr_err("Couldn't delete node '%s', (%d)", name, ret);
-			ret = del_ret;
-		}
+	return deleted;
```
Comment (TonyMouawad) [inline board/miovision/royaloak-smartsense/royaloak-smartsense.c:241]: Can deleted be anything other than 1 or 0?

File: board/miovision/royaloak-smartsense/royaloak-smartsense.c [hunk 2]
```diff
@@ -200,57 +198,65 @@ static void ft_mac_address_setup(void *fdt)
...
-
-	if (txslot == TXSLOT_INVALID) {
-		pr_err("txslot is invalid, skipping board setup\n");
-		return 0;
-	}
+	int ret;
+	int deleted = 0;
 
-	fdt_for_each_subnode(subnode, blob, 0) {
+	fdt_for_each_subnode(subnode, blob, parent) {
```
Comment (TonyMouawad) [inline board/miovision/royaloak-smartsense/royaloak-smartsense.c:211]: If deleted can only be a 1 or a 0, I recommend changing the return type to bool.

File: board/miovision/royaloak-smartsense/royaloak-smartsense.c [hunk 3]
```diff
@@ -200,57 +198,65 @@ static void ft_mac_address_setup(void *fdt)
...
-			pr_err("Couldn't delete node '%s', (%d)", name, ret);
-			ret = del_ret;
-		}
+	return deleted;
+}
+
+int ft_remove_unnecessary_nodes(void *blob, bd_t *bd)
+{
+	int timeout = 50;
+    int ret = 0;
```
Comment (TonyMouawad) [inline board/miovision/royaloak-smartsense/royaloak-smartsense.c]: Fix the indent.

File: board/miovision/royaloak-smartsense/royaloak-smartsense.c [hunk 4]
```diff
@@ -200,57 +198,65 @@ static void ft_mac_address_setup(void *fdt)
...
+		return 0;
+	}
+
+	while (ft_remove_first_unnecessary_node(blob, 0) && timeout > 0) {
+		--timeout;
+	}
+
+	if (timeout <= 0) {
+		pr_err("Timed out on deleting tx2a device tree nodes");
+        ret = 1;
```
Comment (TonyMouawad) [inline board/miovision/royaloak-smartsense/royaloak-smartsense.c]: Fix the indent.

**Review by TonyMouawad — COMMENTED** [review]
Just a couple of tweaks is all.

File: board/miovision/royaloak-smartsense/royaloak-smartsense.c (hunk 2 above)
Comment (m3harvey) [reply board/miovision/royaloak-smartsense/royaloak-smartsense.c:211]: Well I'll be. I didn't know C had a bool type.

File: board/miovision/royaloak-smartsense/royaloak-smartsense.c [hunk 5]
```diff
@@ -200,57 +198,65 @@ static void ft_mac_address_setup(void *fdt)
 	}
 }
 
-static int ft_remove_unnecessary_nodes(void *blob)
+static bool ft_remove_first_unnecessary_node(void *blob, int parent)
```
Comment (TonyMouawad) [inline board/miovision/royaloak-smartsense/royaloak-smartsense.c:202]: Did you not have to #include <stdbool.h>?  Must be included elsewhere, or else this won't compile.

**Review by TonyMouawad — DISMISSED** [review]

File: board/miovision/royaloak-smartsense/royaloak-smartsense.c (hunk 5 above)
Comment (m3harvey) [reply board/miovision/royaloak-smartsense/royaloak-smartsense.c:202]: Good point. Added.

**Review by MikeLoder — APPROVED** [review]

**Review by TonyMouawad — APPROVED** [review]

**Pull request merged**